kernel/Utilities/SparsityEstimator.cpp

```diff
@@ -15,310 +18,416 @@ SparsityEstimator::SparsityEstimator(const GlobalIndexing& rowindexing,
-        if (!face->isOwnedByCurrentProcessor() ||
-            !hasLocalDoFs(face, rowIndexing_.getIncludedUnknowns())) {
-            // If the face is not owned by the current processor, then it
-            // does not have any rows in the matrix on this processor. The
-            // number of non zero entries in the corresponding row does not
-            // need to be estimated on this processor.
-            continue;
-        }
-        workspace.clear();
-        std::vector<const Base::Element*> faceElements = {
-            face->getPtrElementLeft()};
-        if (face->isInternal())
-            faceElements.push_back(face->getPtrElementRight());
-        for (const Base::Element* element : faceElements) {
-            // DoFs from supporting element
-            addElementDoFs(element, workspace);
-            // Face matrix coupling with the element on the other side of the
-            // face
-            if (!includeFaceCoupling) continue;
-            // Face coupling with conforming elements can couple DoFs that
-            // only have shared support on the face between two elements.
-            // For a DoF that is defined on a subdomain boundary face, it
-            // has support on a ghost cell and all the faces of the ghost
-            // cell. As there is only a single layer of ghost cells, we do
-            // in general not have the element on the other side of the
-            // faces of this ghost cell. Therefore, it is not possible to
-            // know how many basis functions have support on this face.
-            //
-            // The only way I can currently see around this is to add
-            // another layer of ghost cells.
-            logger.assert_always(
-                element->isOwnedByCurrentProcessor(),
-                "Face coupling with conforming basis functions on distributed "
-                "meshes is not supported");
-            for (const Base::Face* otherFace : element->getFacesList()) {
-                if (otherFace == face)
-                    continue;  // Both left & right element of the original face
-                               // are already visited
-                if (!otherFace->isInternal()) continue;
-                const Base::Element* otherElement =
-                    otherFace->getPtrOtherElement(element);
-                // The basis function from the face may couple through any other
-                // face of the elements adjacent to the face
-                addElementDoFs(otherElement, workspace);
+                for (Detail::DoFInterval rowInterval : rowIntervals) {
+                    for (Detail::DoFInterval colInterval : ownedColIntervals) {
+                        owned[rowInterval].emplace(colInterval);
+                    }
+                    for (Detail::DoFInterval colInterval :
+                         nonOwnedColIntervals) {
+                        nonOwned[rowInterval].emplace(colInterval);
+                    }
```

Comment:
We append all elements in the for loop, can we not simply append the vectors.

`a.insert(std::end(a), std::begin(b), std::end(b));`

Reply:
Replaced